Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Passthrough.cpp

```diff
@@ -0,0 +1,889 @@
+// SPDX-License-Identifier: MIT
+/*
+$info$
+meta: LinuxSyscalls|syscalls-shared ~ Syscall implementations shared between x86 and x86-64
+tags: LinuxSyscalls|syscalls-shared
+$end_info$
+*/
+
+#include "LinuxSyscalls/Syscalls.h"
+#include "LinuxSyscalls/x64/Syscalls.h"
+#include "LinuxSyscalls/x32/Syscalls.h"
+
+#include <FEXCore/IR/IR.h>
+
+#include <stdint.h>
+#include <sys/epoll.h>
+
+namespace FEX::HLE {
+#ifdef _M_ARM_64
+template<int syscall_num>
+requires (syscall_num != -1)
```

Comment:
Syscall enum values are declared in `Source/Tools/LinuxEmulation/LinuxSyscalls/{Arm64,x32,x64}/SyscallsEnum.h`.
Syscalls that don't exist on that platform are defined as -1.

This catches the programming error of trying to passthrough a syscall that doesn't exist on the host platform and will result in a compile error instead of unexpected ENOSYS.

This was very nice when converting all the passthrough operations and finding ones that were getting passed through when they shouldn't be. Additionally this will be useful in the future when adding new syscalls to ensure we don't accidentally add one that doesn't exist on AArch64. This isn't very frequent but in kernel 6.6 `map_shadow_stack` was added on x86-64 but doesn't exist on AArch64.

Reply:
Ah yes, makes sense. Thanks for the explanation!